.github/workflows/build-pr-green-arm64.yaml

```diff
@@ -207,15 +208,15 @@ jobs:
     env:
       ARCH: arm64
       VAGRANT_CPU: 2
-      VAGRANT_MEMORY: 5120
+      VAGRANT_MEMORY: 6500
```

Comment:
actually... this is a good catch. elemental shouldn't use /tmp by default during upgrades, or it would rely on enough ram to be availlable.. we used to default to COS_PERSISTENT here

Reply:
MEMORY here should even be lower to be realistic. RPI and such don't have so much ram available

Reply:
> VAGRANT_MEMORY here should even be lower to be realistic. RPI and such don't have so much ram available

Not really, this is CI specific. The defaults on the vagrant file should be lowered instead. I put this to 10Gb and 8 cpu on my local host to speed up, nothing to do with the default values. Plus this is for the test machine. Do we want to test like if we were running the tests against an RPI? Because I can already feel the pain just thinking about that lmao

Reply:
no, but having an arm host pretending to have 8gb/10gb of ram is really unreal scenarios... imho we want a test that gives us some guarantees, if we start to pump it to make it work then there is no sense in running it then.

The fact that already made us aware of the elemental-cli used /tmp here is really helpful.. I wouldn't raise the ram here to fix the test

Reply:
no no, the ram is not being raised to fix the test, it is to enable to run and get the full logs on where the hell is elemental thinking of storing the stuff, as locally it uses the proper state dir....so I need to see the full log of this properly to understand the issue as I cant reproduce it. Hopefully now with the full debug log added it will show, plus it can be triggered again as I lowered the mem to 4096

Reply:
ah alright alright gotcha :+1: thought we were bumping ram to make it pass :sweat_smile: looks good then :+1:

Reply:
its a bit weird cause it only happens on arm somehow....

Reply:
interesting that it seems to be luet using the /tmp dir....
```
   INFO    Downloading image  quay.io/costoolkit/releases-green-arm64:cos-system-0.8.9-20
           INFO    Image: quay.io/costoolkit/releases-green-arm64:cos-system-0.8.9-20. Pulled: sha256:083d78ba6786108b31368703f64d8fdd7385febf584674f51f2258e967cfe1cd. Size: 840.3MiB
            ERROR     Failed compressing package quay.io/costoolkit/releases-green-arm64:cos-system-0.8.9-20: write /tmp/tmpluet/artifact596912627.zst: no space left on device
           INFO    Checking for file conflicts..
          panic: runtime error: invalid memory address or nil pointer dereference
          [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x96f4c8]
          
          goroutine 72 [running]:
          github.com/mudler/luet/pkg/api/core/types/artifact.(*Checksums).generateSum(0x400012dcc0, 0x0?, {{0x1120fa8?, 0x401070bc80?}, {0xecc3de?, 0x4000000000?}})
          	/luetbuild/go/pkg/mod/github.com/mudler/luet@v0.0.0-20220127164253-6f77fa2b3a72/pkg/api/core/types/artifact/checksum.go:78 +0x48
          github.com/mudler/luet/pkg/api/core/types/artifact.(*Checksums).generateSHA256(0x0?, 0x0?)
          	/luetbuild/go/pkg/mod/github.com/mudler/luet@v0.0.0-20220127164253-6f77fa2b3a72/pkg/api/core/types/artifact/checksum.go:73 +0x5c
          github.com/mudler/luet/pkg/api/core/types/artifact.(*Checksums).Generate(...)
          	/luetbuild/go/pkg/mod/github.com/mudler/luet@v0.0.0-20220127164253-6f77fa2b3a72/pkg/api/core/types/artifact/checksum.go:60
          github.com/mudler/luet/pkg/api/core/types/artifact.(*PackageArtifact).Verify(0x0)
          	/luetbuild/go/pkg/mod/github.com/mudler/luet@v0.0.0-20220127164253-6f77fa2b3a72/pkg/api/core/types/artifact/artifact.go:112 +0x70
          github.com/mudler/luet/pkg/installer.(*LuetInstaller).getPackage(0x400005c688?, {0x4028b6d7c0?, 0x40086b8bd0?, {0x111d130?, 0x401f740480?}}, {0x112a070?, 0x4028ae8c40?})
          	/luetbuild/go/pkg/mod/github.com/mudler/luet@v0.0.0-20220127164253-6f77fa2b3a72/pkg/installer/installer.go:947 +0x68
          github.com/mudler/luet/pkg/installer.(*LuetInstaller).downloadWorker(0x40000cf280, 0x96d000?, 0x40015ff2d8?, 0x0, 0x16?, {0x112a070, 0x4028ae8c40})
          	/luetbuild/go/pkg/mod/github.com/mudler/luet@v0.0.0-20220127164253-6f77fa2b3a72/pkg/installer/installer.go:981 +0xb4
          created by github.com/mudler/luet/pkg/installer.(*LuetInstaller).download
          	/luetbuild/go/pkg/mod/github.com/mudler/luet@v0.0.0-20220127164253-6f77fa2b3a72/pkg/installer/installer.go:638 +0x3dc
          : Process exited with status 2
```

Reply:
I guess we should set the `LuetConfig.System.TmpDirBase` somewhere but not sure where.... luet default to `os.tmpdir()` any ideas @mudler ?